guides/schema/class_based_api.md

```diff
@@ -68,6 +68,8 @@ As described below, `.to_graphql` can be overridden to customize the type system
 Previously, list types were expressed with `types[T]` and non-null types were expressed with `!T`. Now:
 
 - List types are expressed with Ruby Arrays, `[T]`, for example, `field :owners, [Types::UserType]`
+  - By default, list members are _non-null_, for example, `[Types::UserType]` becomes `[User!]`
+  - If your list members may be null, add `,nil` to the array: `[Types::UserType, nil]` becomes `[User]` (the list may include `nil`)
```

Comment:
this is a clever syntax, :+1:

Reply:
though maybe `[Types::UserType, null: true]` would be even more obvious?

Reply:
i was going to ask ... good clever or bad clever 😅 

I thought about a keyword argument, say, `null_items:`, but the problem there is with nested lists. For example, how could you express `[[[T]!]]`? It's crazy, but I think it happens sometimes 😬

Reply:
I like your suggestion! Actually, I didn't know it was valid Ruby, but voila: 

```ruby
# $ irb
irb(main):001:0> [1, a: true]
# => [1, {:a=>true}]
irb(main):002:0> [[1, a: true], a: false]
# => [[1, {:a=>true}], {:a=>false}]
```

Reply:
Are there any other args that you might want to apply to the inner item but not the field as a whole? That would be a point in favour of `[Type, nested_arg_hash]` since you could add args to the nesting later.

Reply:
Pretty cool, other option is always `[Types::UserType.to_non_null]` or similar but it definitely looks less nice 😁

Reply:
There is very little that *isn't* valid ruby sometimes :P